From 04e110af919047528d08ce344f11ac44e4d8f029 Mon Sep 17 00:00:00 2001
From: Noam Lessner <nlessner@habana.ai>
Date: Sun, 16 Feb 2020 10:29:03 +0200
Subject: [PATCH] [SW-8229] goya2: save pc to stack so we can return

Save blink to stack so when we return from main we can pop
and return to calling function

Change-Id: I39a24478fb569d81e4ce12af566dd7ec8ba52272
Signed-off-by: Noam Lessner <nlessner@habana.ai>
---

diff --git a/app/crt0_ppboot.S b/app/crt0_ppboot.S
index e5d5fd8..b4bd0bb 100644
--- a/app/crt0_ppboot.S
+++ b/app/crt0_ppboot.S
@@ -23,5 +23,11 @@
 	bic	sp, x0, #0xf	/* 16-byte alignment for ABI compliance */
 	mov	x29, sp		/* Set FP (x29) */
 #endif
-
+#ifdef __arc__
+	push_s	blink
+#endif
 	bl	ppboot_main	/* Set LR (x30) */
+#ifdef __arc__
+	pop_s	blink
+	j_s	[blink]
+#endif
